jobs/jobs_api_go_sdk.go

```diff
@@ -156,8 +156,52 @@ func (c controlRunStateLifecycleManagerGoSdk) OnUpdate(ctx context.Context) erro
 	return StopActiveRun(jobID, c.d.Timeout(schema.TimeoutUpdate), w, ctx)
 }
 
-func updateJobClusterSpec(clusterSpec *compute.ClusterSpec, d *schema.ResourceData) error {
-	err := clusters.ModifyRequestOnInstancePool(clusterSpec)
+const (
+	applyPolicyDefaultValuesAllowListField = "__apply_policy_default_values_allow_list"
+)
+
+// This function filters the cluster specification based on the `__apply_policy_default_values_allow_list` field.
+// When this field is configured, only the specified fields are included in the API request.
+// This feature is specifically designed for users leveraging cluster policy defaults with the `apply_policy_default_values` setting.
+//
+// The `__apply_policy_default_values_allow_list` field contains a list of allowed field names within the cluster specification.
+// These field names correspond to the actual fields in the cluster spec structure.
+func observeApplyPolicyDefaultValuesAllowList(d *schema.ResourceData, prefix string, clusterSpec *compute.ClusterSpec) error {
+	// Skip if the `apply_policy_default_values` setting is not enabled.
+	applyPolicyDefaultValues, ok := d.GetOk(prefix + ".apply_policy_default_values")
+	if !ok || !applyPolicyDefaultValues.(bool) {
+		return nil
+	}
+
+	// Skip if the `__apply_policy_default_values_allow_list` is not set or empty.
+	v, ok := d.GetOk(prefix + "." + applyPolicyDefaultValuesAllowListField)
+	if !ok {
+		return nil
+	}
+
+	// Load as a list of strings.
+	var allowList []string
+	for _, v := range v.([]any) {
+		allowList = append(allowList, v.(string))
+	}
```

Comment:
Should we have a check for `v.(string)` to avoid panic?

Reply:
It is guaranteed to be a string per the schema below. I looked for precedent in the codebase and it appears there are more places where we do a hard type assertion for values pulled from `schema.ResourceData`.

I agree it's better to be defensive, but we should update all call sites to use type assertions then.